hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirMkdirOp.java

```diff
@@ -284,8 +294,8 @@ private static INode createDirectoryINode(FSDirectory fsd,
       byte[] component = iip.getPathComponent(i);
       missing[i - existing.length()] =
           createDirectoryINode(fsd, existing, component, perm);
-      missing[i - existing.length()].setParent(parent.asDirectory());
-      parent = missing[i - existing.length()];
+//      missing[i - existing.length()].setParent(parent.asDirectory());
```

Comment:
This relationship is being set on later stage -addSingleDirectory. This check lead to some permission related UT failures.
@shvachko , I need your help to have a check. I just commented it out for reference, if obsolete we shall remove it on later commits.

Reply:
Yes I think it is right to remove those lines since addSingleDirectory() sets up the parent/child relationship.